### PR TITLE
Task #201536 Get AG list based on Aadhaar number

### PR DESCRIPTION
**Pull request by parthgandhitekdi**
Task: [Added API to get list of beneficiaries based on Aadhaar number](https://tracker.tekdi.net/issues/201536)

### CODE REVIEW

File: src/src/beneficiaries/beneficiaries.controller.ts [hunk 1]
```diff
@@ -49,6 +49,21 @@ export class BeneficiariesController {
...
+		@Body() body: Record<string, any>,
+		@Res() response: Record<string, any>,
+	) {
+		const aadhaarNo = body.aadhar_no;
+		const resultPayload =
+			await this.beneficiariesService.byAadhaar(aadhaarNo);
+		return response.status(200).json({
+			success: true,
+			data: resultPayload,
+		});
```
Comment (yogesh-pawar-2023) [inline src/src/beneficiaries/beneficiaries.controller.ts:64]: need to return failure condition

Comment (parthgandhitekdi) [reply src/src/beneficiaries/beneficiaries.controller.ts]: Done

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 2]
```diff
@@ -47,6 +47,42 @@ export class BeneficiariesService {
 		'updated_by',
 	];
 
+	async byAadhaar(aadhaarNo: string) {
+		const beneficiariesByAadhaarQuery = `
+				query MyQuery {
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:52]: fix indent, use one tab with 4 width, for one level of indent

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 3]
```diff
@@ -47,6 +47,42 @@ export class BeneficiariesService {
...
+								id
+								first_name
+								last_name
+								mobile
+							}
+						}
+					}
+				}
+			`;
+		const resultData = (
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:76]: add blank line after query

File: src/src/beneficiaries/beneficiaries.controller.ts [hunk 4]
```diff
@@ -49,6 +49,28 @@ export class BeneficiariesController {
 		return this.beneficiariesService.findAll(request, req, response);
 	}
 
+	@Post('byAadhaar')
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.controller.ts:52]: rename route as

@Post('/admin/list/duplicates-by-aadhhar')

File: src/src/beneficiaries/beneficiaries.controller.ts [hunk 5]
```diff
@@ -49,6 +49,28 @@ export class BeneficiariesController {
...
+	@UseGuards(new AuthGuard())
+	async duplicateByAadhaar(
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.controller.ts:54]: rename as
getBeneficiariesDuplicatesByAadhhar

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 6]
```diff
@@ -47,6 +47,42 @@ export class BeneficiariesService {
 		'updated_by',
 	];
 
+	async byAadhaar(aadhaarNo: string) {
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:50]: rename as
getBeneficiariesDuplicatesByAadhhar

**Review by manojLondhe — CHANGES_REQUESTED** [review]

**Pull request merged**